cpp/src/parquet/metadata.cc

```diff
@@ -922,7 +918,7 @@ class FileMetaDataBuilder::FileMetaDataBuilderImpl {
 
  private:
   const std::shared_ptr<WriterProperties> properties_;
-  std::vector<std::unique_ptr<format::RowGroup>> row_groups_;
```

Comment:
There aren't many benefits of using a pointer of `format::RowGroup` objects in this code.

Reply:
In the case the vector gets expanded, the elements will need to be copied. These objects are probably so small that a copy is not significant in performance.